app/uk/gov/hmrc/helptosave/models/UCResponse.scala

```diff
@@ -16,10 +16,32 @@
 
 package uk.gov.hmrc.helptosave.models
 
-import play.api.libs.json.{Format, Json}
+import play.api.libs.json._
 
-case class UCResponse(ucClaimant: String, withinThreshold: Option[String])
+case class UCResponse(ucClaimant: Boolean, withinThreshold: Boolean)
 
 object UCResponse {
-  implicit val format: Format[UCResponse] = Json.format[UCResponse]
+
+  implicit val reads: Format[UCResponse] = new Format[UCResponse] {
+
+    override def reads(json: JsValue): JsResult[UCResponse] = {
+      ((json \ "ucClaimant").as[String], (json \ "withinThreshold").asOpt[String]) match {
```

Comment:
`.as` will throw if the reading of the field to `String` is unsuccessful. Using `.validate` and `.validateOpt` will allow for better error handling.